src/vm/jitinterface.h

```diff
@@ -294,20 +302,20 @@ class WriteBarrierManager
     WriteBarrierManager();
     void Initialize();
     
-    void UpdateEphemeralBounds(bool isRuntimeSuspended);
-    void UpdateWriteWatchAndCardTableLocations(bool isRuntimeSuspended, bool bReqUpperBoundsCheck);
+    int UpdateEphemeralBounds(bool isRuntimeSuspended);
+    int UpdateWriteWatchAndCardTableLocations(bool isRuntimeSuspended, bool bReqUpperBoundsCheck);
 
 #ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
-    void SwitchToWriteWatchBarrier(bool isRuntimeSuspended);
-    void SwitchToNonWriteWatchBarrier(bool isRuntimeSuspended);
+    int SwitchToWriteWatchBarrier(bool isRuntimeSuspended);
+    int SwitchToNonWriteWatchBarrier(bool isRuntimeSuspended);
 #endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+    size_t GetCurrentWriteBarrierSize();
```

Comment:
This is the most concerning part for me - changing visibility level. Needed for https://github.com/Rattenkrieg/coreclr/blob/refactor-stompwritebarrier-init-path-12586/src/vm/amd64/jitinterfaceamd64.cpp#L725

Reply:
It seems like it would be nice to have start and stop symbols for the write barrier like there are on other platforms, so we can avoid this sort of write barrier information leakage. I don't think you'd need to do that for this PR, though.

Reply:
I'm afraid I can't follow your thoughs here, but since thats not urgent for this PR would you mind submiting separate issue and I will look at that.